lib/web/device_trust_test.go

```diff
@@ -36,85 +36,105 @@ import (
 func TestHandler_DeviceWebConfirm(t *testing.T) {
 	t.Parallel()
 
-	fakeDevices := &fakeDevicesClient{}
-	wPack := newWebPack(
-		t,
-		1, /* numProxies */
-		withDevicesClientOverride(fakeDevices),
-	)
-
-	proxy := wPack.proxies[0]
-	aPack := proxy.authPack(t, "llama", nil /* roles */)
-	webClient := aPack.clt
-
-	ctx := context.Background()
-
-	t.Run("ok", func(t *testing.T) {
-		query := make(url.Values)
-		query.Set("id", "my-token-id")
-		query.Set("token", "my-token-token")
-
-		// Detect client redirects.
-		var redirected bool
-		httpClient := webClient.HTTPClient()
-		httpClient.CheckRedirect = func(req *http.Request, via []*http.Request) error {
-			// Ignore any subsequent redirects that may happen.
-			if redirected {
-				return nil
+	tests := []struct {
+		name               string
+		redirectURI        string
+		expectedRedirectTo string
+	}{
+		{
+			name:               "no redirect_uri",
+			redirectURI:        "",
+			expectedRedirectTo: "/web",
+		},
+		{
+			name:               "with redirect_uri",
+			redirectURI:        "https://example.com/web/custom/path",
+			expectedRedirectTo: "/web/custom/path",
+		},
+		{
+			name:               "with invalid redirect_uri",
+			redirectURI:        "://invalid",
+			expectedRedirectTo: "/web",
+		},
+		{
+			name:               "with external redirect_uri",
+			redirectURI:        "https://example.com/path",
+			expectedRedirectTo: "/web/path",
+		},
+		{
+			name:               "with empty path redirect_uri",
+			redirectURI:        "https://example.com",
+			expectedRedirectTo: "/web",
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+			fakeDevices := &fakeDevicesClient{}
+			wPack := newWebPack(
+				t,
+				1, /* numProxies */
+				withDevicesClientOverride(fakeDevices),
+			)
+			proxy := wPack.proxies[0]
+			aPack := proxy.authPack(t, "llama", nil /* roles */)
+			webClient := aPack.clt
+			ctx := context.Background()
```

Comment:
Could we use a single "pack" for all tests?

Reply:
I'll inspect more on Monday, but moving the packs outside of the tests caused them to fail. I'm its something simple I'm not seeing.